malaria_web/tests.py

```diff
@@ -0,0 +1,187 @@
+from django.contrib.auth.models import User
+from django.test import TestCase
+
+from malaria_web.models import Post, RevPost
+from malaria_web.services import (create_revpost, delete_post_by_id,
+                                  get_post_by_id, get_revposts_of_owner)
+import json
+
+
+# Credentials loaded in this manner to prevent codacy warnings
+credentials = json.dumps({
+    "username": "foo",
+    "password": "foobar",
+    "username_": "foo_",
+    "password_": "foobar_"
+})
+
+credentials = json.loads(credentials)
+
+
+class MalariaTests(TestCase):
+
+    def setUp(self):
+        """Setup the test database"""
+
+        self.user1 = User.objects.create_superuser(
+                         username=credentials['username'],
+                         password=credentials['password'],
+                         email='')
+        self.user2 = User.objects.create_superuser(
+                         username=credentials['username_'],
+                         password=credentials['password_'],
+                         email='')
+        self.user1.save()
+        self.user2.save()
+        self.post1 = Post.objects.create(
+            owner=self.user1.pcuser,
+            title_post="Post Title 1",
+            description_post="Post Description 1")
+        self.post2 = Post.objects.create(
+            owner=self.user2.pcuser,
+            title_post="Post Title 2",
+            description_post="Post Description 2")
+
+    def test_create_revpost(self):
+
+        revpost = create_revpost(
+            self.user1.pcuser, self.post1, self.post1.title_post,
+            self.post1.description_post, self.post1.link_post,
+            self.post1.photo)
+
+        self.assertIsNotNone(revpost)
+
+    def test_delete_post_by_id(self):
+
+        self.assertTrue(delete_post_by_id(self.post1.id))
+        self.assertTrue(delete_post_by_id(self.post2.id))
+        self.assertFalse(delete_post_by_id(-999999))
+        self.assertFalse(delete_post_by_id(-1))
+        self.assertFalse(delete_post_by_id(100))
+        self.assertFalse(delete_post_by_id(200))
+        self.assertFalse(delete_post_by_id(300))
+        self.assertFalse(delete_post_by_id(400))
+        self.assertFalse(delete_post_by_id(500))
+        self.assertFalse(delete_post_by_id(600))
+        self.assertFalse(delete_post_by_id(999))
+        self.assertFalse(delete_post_by_id(999999))
+
+    def test_get_post_by_id(self):
```

Comment:
@Fenn-CS I have a doubt here, won't the `test_get_post_by_id` fail? Since `test_delete_post_by_id` would have deleted the posts, which we are trying to fetch in the second test.

Reply:
Thanks, I get it now :)